cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -134,10 +134,15 @@ public void getReports(HttpServiceRequest request, HttpServiceResponder responde
       // the number of report directories or the list is reaching the given limit
       while (idx < reportIdDirs.size() && reportStatuses.size() < limit) {
         Location reportIdDir = reportIdDirs.get(idx++);
+        ReportStatus status = getReportStatus(reportIdDir);
         String reportId = reportIdDir.getName();
         // Report ID is time based UUID. Get the creation time from the report ID.
         long creationTime = ReportIds.getTime(reportId, TimeUnit.SECONDS);
-        reportStatuses.add(new ReportStatusInfo(reportId, creationTime, getReportStatus(reportIdDir)));
+        // Read the report request from _START file, which was written at the beginning of report generation
+        String reportRequestString =
+          new String(ByteStreams.toByteArray(reportIdDir.append(START_FILE).getInputStream()), StandardCharsets.UTF_8);
+        ReportGenerationRequest reportRequest = GSON.fromJson(reportRequestString, REPORT_GENERATION_REQUEST_TYPE);
+        reportStatuses.add(new ReportStatusInfo(reportId, reportRequest.getName(), null, creationTime, null, status));
```

Comment:
well if description was given in request, it will not be null right? why are you ignoring it ?

Reply:
also the report status should be sorted by creation time in descending oder.

Reply:
the description will not be given in the report generation request, it will only be given when the report is saved